test_run.lua

```diff
@@ -284,6 +285,26 @@ local function grep_log(self, node, what, bytes, opts)
     return found
 end
 
+-- Checks fn with delay until timeout or fn returns true, otherwise returns false
+local function wait_cond(self, fn, timeout, delay)
+    assert(timeout ~= nil)
+    delay = delay or 0.001
+    local t1 = clock.monotonic()
+    while not fn() do
+        local work_time = clock.monotonic() - t1
+        if work_time > timeout then
+            return false
+        end
+        fiber.sleep(delay)
+    end
+    return true
+end
+
+-- Wrapper for grep_log, wait until expected log entry is appear in server log file
+local function wait_log(self, node, what, bytes, timeout, opts)
+    return wait_cond(self, function() return grep_log(self, node, what, bytes, opts) ~= nil end, timeout)
```

Comment:
I think we should not discard the value returned from grep_log, just bind needed arguments (perform carrying of the function). Also we can allow a user to set delay. Proposed (didn't tested):

```
local delay = opts.delay
opts.delay = nil
local cond = function()
    return grep_log(self, node, what, bytes, opts)
end
return wait_cond(self, cond, timeout, delay)
```